lib/ProductOpener/APIProductWrite.pm

```diff
@@ -186,13 +187,76 @@ sub update_packagings ($request_ref, $product_ref, $field, $is_addition, $value)
 	return;
 }
 
-=head2 update_product_fields ($request_ref, $product_ref)
+=head2 update_tags_fields ($request_ref, $product_ref, $field, $is_addition, $value)
+
+Update packagings.
+
+=cut
+
+sub update_tags_fields ($request_ref, $product_ref, $field, $is_addition, $tags_lc, $value) {
+
+	my $request_body_ref = $request_ref->{body_json};
+	my $response_ref = $request_ref->{api_response};
+
+	if (ref($value) ne 'ARRAY') {
+		add_error(
+			$response_ref,
+			{
+				message => {id => "invalid_type_must_be_array"},
+				field => {id => $field},
+				impact => {id => "field_ignored"},
+			}
+		);
+	}
+	else {
+		# Generate a comma separated list of tags, so that we can use existing functions to add tags
+		my $tags_list = join(',', @$value);
```

Comment:
Today I saw a products with many times the same brand, it disapeared once I edited. But I think a fool-proof mechanism (and a warning) to remove duplicates would be good.

Reply:
see https://github.com/openfoodfacts/openfoodfacts-server/issues/8355